src/rt/gtfsrt_update.cc

```diff
@@ -419,11 +441,149 @@ bool update_run(source_idx_t const src,
+              : std::span{rtt_const.rt_transport_location_seq_[r.rt_]};
+
+      // add rt_transport if not existent
+      if (!r.is_rt()) {
+        r.rt_ = rtt.add_rt_transport(src, tt, r.t_);
+      }
+
+      // match position to stop
+      auto const stop_it = utl::find_if(location_seq, [&](auto const& loc) {
+        auto const stop_position =
+            tt.locations_.get(location_idx_t{loc & 0x0FFFFFFF});
+        auto const vehicle_position = geo::latlng{vp_lat, vp_lon};
+        return geo::approx_squared_distance(
+                   stop_position.pos_, vehicle_position,
+                   geo::approx_distance_lng_degrees(vehicle_position)) < 10;
+      });
+      if (stop_it == end(location_seq)) {
+        log(log_lvl::info, "rt.gtfs.unsupported",
+            R"(Position of Vehicle was not near stop. Skipping Message)", tag,
+            entity.id());
+        ++stats.vehicle_position_position_not_at_stop_;
+        continue;
+      }
+
+      // check for outdated or illegal timestamp
+      auto const vp_ts_utt = unixtime_t{std::chrono::duration_cast<i32_minutes>(
+          std::chrono::seconds{vp.timestamp()})};
+      if ((rtt.rt_transport_latest_vehicle_time_.contains(r.rt_)
+               ? rtt.rt_transport_latest_vehicle_time_[r.rt_] > vp_ts_utt
+               : tt.event_time(r.t_, 0, event_type::kDep) > vp_ts_utt) ||
+          vp_ts_utt > std::chrono::time_point_cast<i32_minutes>(
+                          std::chrono::system_clock::now())) {
+        log(log_lvl::info, "rt.gtfs.unsupported",
+            R"(Outdated or illegal timestamp. Skipping Message)", tag,
+            entity.id());
+        ++stats.outdated_or_illegal_vehicle_position_timestamps_;
+        continue;
+      }
+
+      // check if stop is "new" or the first one in the sequence
+      auto const stopped_at_idx = stop_idx_t(std::distance(begin(location_seq), stop_it));
+      if ((rtt.rt_transport_latest_stop_.contains(r.rt_) &&
+           rtt.rt_transport_latest_stop_[r.rt_] > stopped_at_idx) ||
+          stopped_at_idx == 0) {
+        log(log_lvl::info, "rt.gtfs.unsupported",
+            R"(Illegal stop. Skipping Message)", tag, entity.id());
+        ++stats.illegal_stop_;
+        continue;
+      }
+
+      rtt.rt_transport_latest_vehicle_time_[r.rt_] = vp_ts_utt;
+      rtt.rt_transport_latest_stop_[r.rt_] = stopped_at_idx;
+
+      // get remaining stops
+      auto const fr = r.is_scheduled() ? frun::from_t(tt, &rtt_const, r.t_)
+                                       : frun::from_rt(tt, &rtt_const, r.rt_);
+      auto const seq_numbers = fr.stop_range_;
+
+      // get delay
+      auto const et = tt.event_time(r.t_, stopped_at_idx, event_type::kArr);
+      auto const delay = vp_ts_utt - et;
+      auto const delay_cast = std::chrono::duration_cast<duration_t>(delay);
+
+      // update delay for remaining stops
+      auto seq_it = begin(seq_numbers);
+      for (int i = 0; i <= stopped_at_idx; ++i) {
+        ++seq_it;
```

Comment:
how is it 100% sure that now the last stop before doesn't have a time >= the first updated stop? can't we check this here and just propagate along the whole trip instead of using large hash_maps that waste RAM?